src/modules/urlShortener/urlOperations.js

```diff
@@ -0,0 +1,161 @@
+		- In case of a collision, the first 6 characters will be a substring of the base64 hash
+		  and the 7th character will be appended from the chars string in a sequential manner.
+		- In case of getShortURL api call, simply hash and try to insert it to the DB.
+		- If everything goes well it means we have successfully created the short url.
+		- In case of an error there are two possibilities:
+		  - There is a collision.
+		  - The long url already exists.
+		- Do a LIKE query with the (0..6) substring of the hash.
+		- On getting back results, check whether the long URL of the results returned from
+		  the DB matches the long URL passed as parameter.
+		- If any of them does, return the corresponding short URL.
+		- In any other case, we have a collision. We can resolve it by appending 7th character which
+		  is 1 place higher than the highest 7th position character of the short URLS from the result.
+*/
+
+const chars = '-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz';
+const doReadQuery = promisify(pg.read.bind(pg, config.connStr));
+const doWriteQuery = promisify(pg.write.bind(pg, config.connStr));
+
+const isShortURL = (shortURL: string): boolean => {
+	if (shortURL.length === 6 || shortURL.length === 7) {
+		return /^[A-Za-z0-9\-_]+$/.test(shortURL);
+	}
+	return false;
+};
+
+const getPathFromURL = (longURL: string): string => {
+	return longURL.replace(/https?:\/\/[^\/]+/, '');
+};
+
+const makeURLSafeHash = (pathFromLongURL: string): string => {
+	const md5Hash = crypto.createHash('md5').update(pathFromLongURL).digest('base64');
+	// regex to make url safe hash, converts '+' to '-', '/' to '_', removes "=" from the end
+	const urlSafeHash = md5Hash.replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '');
+	return urlSafeHash.slice(0, 6);
+};
+
+const insertLongURL = (shortURL: string, longURL: string): Promise<Array<{rowCount: number}>> => {
+	return doWriteQuery([ {
+		$: `INSERT INTO urls (
+			shorturl, longurl
+		) VALUES (
+			&{shortURL}, &{longURL}
+		)`,
+		shortURL,
+		longURL
+	} ]);
+};
+
+export const getShortURL = (longURL: string): Promise<string> => {
+	const pathFromLongURL = getPathFromURL(longURL);
+	const shortURL = makeURLSafeHash(pathFromLongURL);
+	// Try inserting long URL and its hash as the short URL.
+	return insertLongURL(shortURL, pathFromLongURL)
+		// in case of a successful insertion, simply return the hash.
+		.then((result) => {
+			// result: [{.., rowcount: _, ...}, ..]
+			if (result[0].rowCount === 1) {
+				winston.info('Insert operation performed successfully.');
+				return shortURL;
+			}
+			return null;
+		})
+		// in case of an error, take proper measures.
+		.catch((error) => {
+			winston.error(`something went wrong while perfoming insertion, ${error}`);
+			return doReadQuery({
+				$: 'SELECT shorturl, longurl FROM urls WHERE shorturl LIKE &{shortURL}',
+				shortURL: shortURL + '%'
+			})
+			.then((urlsList) => {
```

Comment:
{ shorturl, longurl } <-- // link
links
for (link of links)
link.shorturl, link.longurl